packages/evm/src/eof/eof.ts

```diff
@@ -0,0 +1,327 @@
+      }
+      maxHeight = Math.max(maxHeight, nextStackHeight)
+      if (maxHeight > 1023) {
+        throw new Error('Stack height exceeds the maximum of 1024')
+      }
+
+      // Special case; for RJUMP do not immediately continue (need to check jump location and add intermediate bytes)
+      if (opcode !== 0x5c && terminatingOpcodes.has(opcode)) {
+        if (
+          opcode === 0xb1 &&
+          currentStackHeight !== container.body.typeSections[currentSection].outputs
+        ) {
+          // RETF stack height is invalid
+          throw new Error('RETF stack height invalid')
+        }
+        continue
+      }
+
+      const nextPcs = []
+      if (opcode >= 0x60 && opcode <= 0x7f) {
+        // PUSH
+        nextPcs.push(pc + 1 + opcode - 0x5f)
+        for (let i = 1; i <= opcode - 0x5f; i++) {
+          visitedSet.add(pc + i)
+        }
+      } else if (opcode === 0x5c) {
+        // RJUMP
+        nextPcs.push(pc + 3 + code.readInt16BE(pc + 1))
+        visitedSet.add(pc + 1)
+        visitedSet.add(pc + 2)
+      } else if (opcode === 0x5d) {
+        // RJUMPI
+        nextPcs.push(pc + 3)
+        nextPcs.push(pc + 3 + code.readInt16BE(pc + 1))
+        visitedSet.add(pc + 1)
+        visitedSet.add(pc + 2)
+      } else if (opcode === 0x5e) {
+        // RJUMPV
+        const items = code[pc + 1]
+        visitedSet.add(pc + 1)
+        const finalPos = pc + 2 + items * 2
+        for (let i = 0; i < items; i++) {
+          nextPcs.push(finalPos + code.readInt16BE(pc + 2 + i * 2))
+          visitedSet.add(pc + 2 + i * 2)
+          visitedSet.add(pc + 3 + i * 2)
+        }
+        nextPcs.push(finalPos)
+      } else if (opcode === 0xb0) {
+        const functionToCall = code.readUint16BE(pc + 1)
+        if (currentStackHeight < container.body.typeSections[functionToCall].inputs) {
+          // stack underflow on function call
+          throw new Error('Cannot CALLF: stack underflow')
+        }
+        visitedSet.add(pc + 1)
+        visitedSet.add(pc + 2)
+        // CALLF
+        nextPcs.push(pc + 3)
+      } else {
+        nextPcs.push(pc + 1)
+      }
+
+      for (const nextPc of nextPcs) {
+        if (stackHeight[nextPc] !== undefined) {
+          if (stackHeight[nextPc] !== nextStackHeight) {
+            throw new Error(
+              'Stack height target is different than previous target (stack loop error)'
+            )
+          }
+        } else {
+          stackHeight[nextPc] = nextStackHeight
+          worklist.push(nextPc)
+        }
+      }
+    }
+    if (visitedSet.size !== code.length) {
+      throw new Error('There exist opcodes which cannot be reached')
+    }
+    if (maxHeight !== container.body.typeSections[currentSection].maxStackHeight) {
+      throw new Error('Section max height does not correspond to actual max height')
+    }
+  }
+
+  return true
+}
+
+// Deprecate below opcodes (this is breaking, but we should remove those)
```

Comment:
What is this referencing?

Reply:
This should be removed. The EOF changes are breaking, but since the spec has been overhauled I think this is OK. (And I also doubt that people were using our EOF version of March)